### PR TITLE
[JENKINS-64038] environment variables are Strings

### PR DESCRIPTION
**Pull request by jtnord**
The initial build is not splitting correctly.

this attempts to fix that based on a guess that the if is failing as it
is comparing a String to an int.

### CODE REVIEW

**Comment by jtnord** [comment]
image

this was the cause of the first batch not splitting.

This still has the issue that the excludes are not correctly taken into account.

File: Jenkinsfile [hunk 1]
```diff
@@ -8,7 +8,7 @@ for (int i = 0; i < (BUILD_NUMBER as int); i++) {
 def branches = [:]
 
 def splits
-if (env.BUILD_NUMBER == 1) {
+if (env.BUILD_NUMBER == '1') {
```
Comment (jglick) [inline Jenkinsfile:11]: or
```suggestion
if (BUILD_NUMBER == '1') {
```

**Review by jglick — APPROVED** [review]

**Comment by jtnord** [comment]
marked as do not merge until we can also find out why the splits are not being excluded in surefire.
if merged before that all first builds of PRs will squat on the nodes running all tests 20 times (10 times for each of Java8/java11)

**Comment by jtnord** [comment]
removed the do-not-merge the splitting seems to be working ok "most of the time".  attempts to work out why it was not always working are inconclusive.
so lets just merge this and keep an eye on things.

**Pull request merged**